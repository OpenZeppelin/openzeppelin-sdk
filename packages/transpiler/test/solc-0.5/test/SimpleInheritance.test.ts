import { shouldTranspileToValidContract } from './LocalConract.behaviour';

describe(`SimpleInheritance contract`, (): void => {
  shouldTranspileToValidContract('SIC', {
    SIC: {
      path: 'SimpleInheritance',
      fileName: 'SimpleInheritance',
      contracts: ['SIC', 'SIB', 'SIA'],
    },
    SIB: {
      path: 'SimpleInheritance',
      fileName: 'SimpleInheritance',
      contracts: ['SIB', 'SIA'],
    },
    SIA: {
      path: 'SimpleInheritance',
      fileName: 'SimpleInheritance',
      contracts: ['SIA'],
    },
  });
});
