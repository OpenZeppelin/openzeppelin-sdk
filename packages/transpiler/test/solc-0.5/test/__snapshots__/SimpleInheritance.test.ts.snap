// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleInheritance contract is converted to a valid SICUpgradable contract 1`] = `
"pragma solidity ^0.5.0;
import \\"@openzeppelin/upgrades/contracts/Initializable.sol\\";

contract SIAUpgradable is Initializable {
    function initialize() external initializer {
        __init(true);
    }

    function __init(bool callChain) internal {
        
        
        
        emit log(\\"SIA\\");
    
    }

    uint256 public foo;
    event log(string);
    
}

contract SIBUpgradable is Initializable, SIAUpgradable {
    function initialize() external initializer {
        __init(true);
    }

    function __init(bool callChain) internal {
        if(callChain) {
            SIAUpgradable.__init(false);
        }
        
        val = 123;
        
    }

    uint256 public val ;
}

contract SICUpgradable is Initializable, SIBUpgradable {
    function initialize() external initializer {
        __init(true);
    }

    function __init(bool callChain) internal {
        if(callChain) {
            SIAUpgradable.__init(false);
            SIBUpgradable.__init(false);
        }
        
        bar = \\"hello\\";
        
        bar = \\"changed\\";
        emit log(\\"SIC\\");
    
    }

    string public bar ;
    
}"
`;
